app/src/migration/java/org/mozilla/fenix/MigrationPushSubscriber.kt

```diff
@@ -0,0 +1,42 @@
+package org.mozilla.fenix
+
+import android.content.Context
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.flow.collect
+import mozilla.components.concept.push.PushService
+import mozilla.components.lib.state.ext.flowScoped
+import mozilla.components.support.migration.state.MigrationProgress
+import mozilla.components.support.migration.state.MigrationStore
+
+/**
+ * Migration-aware subscriber that disables the push service during an active migration
+ * and re-enables when complete.
+ */
+class MigrationPushSubscriber(
+    private val context: Context,
+    private val service: PushService,
+    private val store: MigrationStore
+) {
+    private var scope: CoroutineScope? = null
+
+    @UseExperimental(ExperimentalCoroutinesApi::class)
```

Comment:
any way to avoid using this? i'm assuming it's for `flowScoped`?

or - is this an established pattern we have elsewhere in the code already?

Reply:
Yes, we do this [everywhere](https://searchfox.org/mozilla-mobile/search?path=&q=ExperimentalCoroutinesApi). I added it to the method only here since that's the only call site.